src/Util/FileHelper.php

```diff
@@ -14,12 +15,21 @@ public static function normalizeWindowsPath(string $path) : string
         return str_replace('\\', '/', $path);
     }
 
-    public static function normalizeSystemPath(string $path) : string
+    public static function normalizeSystemPath(string $originalPath, string $directorySeparator = DIRECTORY_SEPARATOR) : string
     {
-        $path = self::normalizeWindowsPath($path);
+        $originalPath = self::normalizeWindowsPath($originalPath);
+        preg_match('~^([a-z]+)\\:\\/\\/(.+)~', $originalPath, $matches);
+        if ($matches !== []) {
+            [, $scheme, $path] = $matches;
+        } else {
+            $scheme = null;
+            $path   = $originalPath;
+        }
 
-        return DIRECTORY_SEPARATOR === '\\'
-            ? str_replace('/', '\\', $path)
-            : $path;
+        return ($scheme !== null ? $scheme . '://' : '') . (
+            $directorySeparator === '\\'
```

Comment:
I don't think `$directorySeparator` should leak into the API of this helper: this shifts responsibility to the caller, and that's only more trouble.

Reply:
@Ocramius I did this so that the code is easily testable, no matter the platform. Any suggestion what to do instead?

Reply:
Hardcoding: we already have CI running against both linux and windows 👍

Reply:
Alright, pushed some changes